cheetah/accelerator/transverse_deflecting_cavity.py

```diff
@@ -0,0 +1,251 @@
+        y = incoming.particles[..., 2]
+        py = incoming.particles[..., 3]
+        tau = incoming.particles[..., 4]
+        delta = incoming.particles[..., 5]
+
+        z, pz, p0c = bmadx.cheetah_to_bmad_z_pz(tau, delta, incoming.energy, mc2)
+
+        x_offset = self.misalignment[..., 0]
+        y_offset = self.misalignment[..., 1]
+
+        # Begin Bmad-X tracking
+        x, px, y, py = bmadx.offset_particle_set(
+            x_offset, y_offset, self.tilt, x, px, y, py
+        )
+
+        x, y, z = bmadx.track_a_drift(self.length / 2, x, px, y, py, z, pz, p0c, mc2)
+
+        voltage = self.voltage / p0c
+        k_rf = 2 * torch.pi * self.frequency / c_light
+        phase = (
+            2
+            * torch.pi
+            * (self.phase - (bmadx.particle_rf_time(z, pz, p0c, mc2) * self.frequency))
+        )
+
+        px = px + voltage * torch.sin(phase)
+
+        beta = (1 + pz) * p0c / torch.sqrt(((1 + pz) * p0c) ** 2 + mc2**2)
+        beta_old = beta
+        E_old = (1 + pz) * p0c / beta_old
+        E_new = E_old + voltage * torch.cos(phase) * k_rf * x * p0c
+        pc = torch.sqrt(E_new**2 - mc2**2)
+        beta = pc / E_new
+
+        pz = (pc - p0c) / p0c
+        z = z * beta / beta_old
+
+        x, y, z = bmadx.track_a_drift(self.length / 2, x, px, y, py, z, pz, p0c, mc2)
+
+        x, px, y, py = bmadx.offset_particle_unset(
+            x_offset, y_offset, self.tilt, x, px, y, py
+        )
+        # End of Bmad-X tracking
+
+        # Convert back to Cheetah coordinates
+        tau, delta, ref_energy = bmadx.bmad_to_cheetah_z_pz(z, pz, p0c, mc2)
+
+        outgoing_beam = ParticleBeam(
+            torch.stack((x, px, y, py, tau, delta, torch.ones_like(x)), dim=-1),
+            ref_energy,
+            particle_charges=incoming.particle_charges,
+            device=incoming.particles.device,
+            dtype=incoming.particles.dtype,
+        )
+        return outgoing_beam
+
+    def broadcast(self, shape: Size) -> Element:
+        return self.__class__(
+            length=self.length.repeat(shape),
+            voltage=self.voltage.repeat(shape),
+            phase=self.phase.repeat(shape),
+            frequency=self.frequency.repeat(shape),
+            misalignment=self.misalignment.repeat((*shape, 1)),
+            tilt=self.tilt.repeat(shape),
+            tracking_method=self.tracking_method,
+            name=self.name,
+            device=self.length.device,
+            dtype=self.length.dtype,
+        )
+
+    def split(self, resolution: torch.Tensor) -> list[Element]:
+        # TODO: Implement splitting for cavity properly, for now just returns the
+        # element itself
+        return [self]
+
+    def plot(self, ax: plt.Axes, s: float) -> None:
+        alpha = 1 if self.is_active else 0.2
+        height = 0.4
+
+        patch = Rectangle(
+            (s, 0), self.length[0], height, color="gold", alpha=alpha, zorder=2
+        )
+        ax.add_patch(patch)
```

Comment:
Is it bad that `Cavity` and `TransverseDeflectingCavity` currently plot the same?

Reply:
I would suggest to use a different color.